models/metadata.go

```diff
@@ -0,0 +1,83 @@
+	Contacts          []ContactDetails     `json:"contacts,omitempty"`
+	Description       string               `json:"description,omitempty"`
+	Dimensions        []CodeList           `json:"dimensions,omitempty"`
+	Downloads         *DownloadList        `json:"downloads,omitempty"`
+	Keywords          []string             `json:"keywords,omitempty"`
+	LatestChanges     *[]LatestChange      `json:"latest_changes,omitempty"`
+	License           string               `json:"license,omitempty"`
+	Links             *MetadataLinks       `json:"links,omitempty"`
+	Methodologies     []GeneralDetails     `json:"methodologies,omitempty"`
+	NationalStatistic *bool                `json:"national_statistic,omitempty"`
+	NextRelease       string               `json:"next_release,omitempty"`
+	Publications      []GeneralDetails     `json:"publications,omitempty"`
+	Publisher         *Publisher           `json:"publisher,omitempty"`
+	QMI               *GeneralDetails      `json:"qmi,omitempty"`
+	RelatedDatasets   []GeneralDetails     `json:"related_datasets,omitempty"`
+	ReleaseDate       string               `json:"release_date,omitempty"`
+	ReleaseFrequency  string               `json:"release_frequency,omitempty"`
+	Temporal          *[]TemporalFrequency `json:"temporal,omitempty"`
+	Theme             string               `json:"theme,omitempty"`
+	Title             string               `json:"title,omitempty"`
+	UnitOfMeasure     string               `json:"unit_of_measure,omitempty"`
+	URI               string               `json:"uri,omitempty"`
+}
+
+type MetadataLinks struct {
+	AccessRights *LinkObject `json:"access_rights,omitempty"`
+	Self         *LinkObject `json:"self,omitempty"`
+	Spatial      *LinkObject `json:"spatial,omitempty"`
+	Version      *LinkObject `json:"version,omitempty"`
+}
+
+func CreateMetaDataDoc(datasetDoc *Dataset, versionDoc *Version) *Metadata {
+	metaDataDoc := &Metadata{
+		Alerts:            versionDoc.Alerts,
+		Contacts:          datasetDoc.Contacts,
+		Description:       datasetDoc.Description,
+		Dimensions:        versionDoc.Dimensions,
+		Downloads:         versionDoc.Downloads,
+		Keywords:          datasetDoc.Keywords,
+		LatestChanges:     versionDoc.LatestChanges,
+		Links:             &MetadataLinks{},
+		License:           datasetDoc.License,
+		Methodologies:     datasetDoc.Methodologies,
+		NationalStatistic: datasetDoc.NationalStatistic,
+		NextRelease:       datasetDoc.NextRelease,
+		Publications:      datasetDoc.Publications,
+		Publisher:         datasetDoc.Publisher,
+		QMI:               datasetDoc.QMI,
+		RelatedDatasets:   datasetDoc.RelatedDatasets,
+		ReleaseDate:       versionDoc.ReleaseDate,
+		ReleaseFrequency:  datasetDoc.ReleaseFrequency,
+		Temporal:          versionDoc.Temporal,
+		Theme:             datasetDoc.Theme,
+		Title:             datasetDoc.Title,
+		UnitOfMeasure:     datasetDoc.UnitOfMeasure,
+		URI:               datasetDoc.URI,
```

Comment:
might be worth checking this URI - not neccesarily for this PR, but should this actually be a link to the version/be called something else. maybe a chat with rob/darren

Reply:
Most likely, relates to the dataset landing page. If it does change then we need to refactor the dataset endpoints